src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -157,11 +158,20 @@ errors::eType TNumberSchemaItem<NumberType>::validate(
     report->set_validation_info(validation_info);
     return errors::INVALID_VALUE;
   }
+
   NumberType value(0);
   if (typeid(int32_t) == typeid(value)) {
-    value = utils::SafeStaticCast<int64_t, int32_t>(Object.asInt());
+    if (Object.asInt() < std::numeric_limits<int32_t>::min() ||
+        Object.asInt() > std::numeric_limits<int32_t>::max()) {
+      return errors::OUT_OF_RANGE;
+    }
+    value = Object.asInt();
   } else if (typeid(uint32_t) == typeid(value)) {
-    value = utils::SafeStaticCast<uint64_t, uint32_t>(Object.asUInt());
+    if (Object.asInt() < std::numeric_limits<uint32_t>::min() ||
+        Object.asInt() > std::numeric_limits<uint32_t>::max()) {
+      return errors::OUT_OF_RANGE;
```

Comment:
```suggestion
      std::string validation_info = "Value out of range";
      report->set_validation_info(validation_info);
      return errors::OUT_OF_RANGE;
```
Probably best to include error info here

Reply:
@jacobkeeler good point. Added in https://github.com/smartdevicelink/sdl_core/pull/3750/commits/e1412cc9e6773f70b999805507dbbcba38150fd0